### PR TITLE
#21 first number.power try

### PR DESCRIPTION
**Pull request by maxonfjvipon**
@yegor256 please take a look and let me know what you think. It's not perfect, just first try, it causes StackOverflow error on big numbers. 
Also could you please try to answer why `power-of-nan` test works, but `power-nan` doesn't?
image

### CODE REVIEW

**Comment by yegor256** [comment]
@Graur please take a look

**Comment by Graur** [comment]
@maxonfjvipon I think `power-nan` test fails when you trying to get `as-float` from `number nan`.

File: src/main/eo/org/eolang/math/number.eo [hunk 1]
```diff
@@ -23,6 +23,9 @@
 +package org.eolang.math
 +home https://github.com/objectionary/eo-math
 +version 0.0.0
++alias org.eolang.math.positive-infinity
++alias org.eolang.math.nan
++alias org.eolang.switch
 +rt jvm org.eolang:eo-math:0.0.0
 
 [n] > number
```
Comment (Graur) [inline src/main/eo/org/eolang/math/number.eo:31]: @maxonfjvipon I guess `number` should decorate `n`, e.g. 
```
[n] > number
  n > @
```

File: src/main/eo/org/eolang/math/number.eo [hunk 2]
```diff
@@ -155,7 +158,41 @@
         value.neg
 
   # Make ^.n power x
-  [x] > pow /?
+  [x] > power
```
Comment (Graur) [inline src/main/eo/org/eolang/math/number.eo:161]: @maxonfjvipon I think we should not remove the `pow` object until the `power` object is stable in use.

**Review by Graur — CHANGES_REQUESTED** [review]
@maxonfjvipon Good one! Just a few comments below.

**Comment by maxonfjvipon** [comment]
> @maxonfjvipon I think `power-nan` test fails when you trying to get `as-float` from `number nan`. 

But I also do `as-float > self-as-float!` and it doesn't fail in `power-of-nan` test. This one is weird

**Comment by maxonfjvipon** [comment]
@Graur they both should end on the first case. Idk why but `(number x).is-nan` returns false? But ok, I'll figure it out
image

**Comment by Graur** [comment]
@maxonfjvipon `switch` object dataize all case conditions, whether they are true or not. One by one. So three of four case conditions dataize `x-as-float` object. And no one dataize `self-as-float`.

**Comment by maxonfjvipon** [comment]
@Graur Got you, thanks, that means a lot!

**Comment by maxonfjvipon** [comment]
@Graur I think it's good to merge, please take a look one more time

File: src/main/eo/org/eolang/math/number.eo (hunk 1 above)
Comment (maxonfjvipon) [reply src/main/eo/org/eolang/math/number.eo:31]: done

File: src/main/eo/org/eolang/math/number.eo (hunk 2 above)
Comment (maxonfjvipon) [reply src/main/eo/org/eolang/math/number.eo:163]: done

**Comment by Graur** [comment]
> @Graur Got you, thanks, that means a lot! BTW: it's one more point to [this ticket](https://github.com/objectionary/eo/issues/774) about double object implementation. We could make `as-float` in this way (or something like this):
> 
> ```
> [] > as-float
>   if. > @
>     is-nan
>     nan
>     /float
> ```

@maxonfjvipon Good point, I agree. Can you implement?

**Review by Graur — APPROVED** [review]

**Comment by Graur** [comment]
> @Graur I think it's good to merge, please take a look one more time

@maxonfjvipon Looks good to me! 
@yegor256 what do you think?

**Comment by yegor256** [comment]
@maxonfjvipon this means that `power` will always return `float`?

**Comment by maxonfjvipon** [comment]
> @maxonfjvipon this means that `power` will always return `float`?

For now - yes. In future I want to add logic for raising number to a FLOAT power, for ex: `(number 4).power 0.5`. It requires some extra logic and `sqrt` object. And then I'll split `powf` to `powf` that will return float and `powi` that will return int

**Pull request closed without merging**